woocommerce/woocommerce
# Render GTIN in Product structured data

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR adds the GTIN data into the product structured data. Only if its a valid number

Ref: pffQ75-pV-p2#comment-478

Related to https://github.com/woocommerce/woocommerce/pull/49312



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to a Product and add a correct GTIN in the inventory section. For example: 012345678905
2. Go to the product and inspect the page. Search for `gtin` in the source. See it's present in the structured data.
3. Go back to the product. Save the GTIN with the wrong format. For example ABC-123
4.  Go to the product and inspect the page. Search for `gtin` in the source.See you cannot find it in the structured data.

<img width="516" alt="Screenshot 2024-07-29 at 17 38 17" src="https://github.com/user-attachments/assets/549316a2-fff3-4d70-8236-bce19adefa2f">


https://github.com/user-attachments/assets/e60e810d-0f77-40e7-9105-ebc13cacdfe4



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
